content/blog/2021-02-18-dvc-2-0-pre-release.md

```diff
@@ -0,0 +1,562 @@
+5649f62d845fdc29e28ea6f7672dd729d3946940 refs/exps/71/67904d89e116f28daf7a6e4c0878268117c893/exp-80655
+f16e7b7c804cf52d91d1d11850c15963fb2a8d7b refs/exps/97/d69af70c6fb4bc59aefb9a87437dcd28b3bde4/exp-63ee0
+0566d42cddb3a8c4eb533f31027f0febccbbc2dd refs/exps/91/94265d5acd847e1c439dd859aa74b1fc3d73ad/exp-bb55c
+9bb067559583990a8c5d499d7435c35a7c9417b7 refs/exps/49/5c835cd36772123e82e812d96eabcce320f7ec/exp-9bf22
+```
+
+The best experiment can be promoted to the workspace and commited to Git.
+
+```dvc
+$ dvc exp apply exp-bb55c
+$ git add .
+$ git commit -m 'optimize max feature size'
+```
+
+Alternatively, an experiment can be promoted to a branch (`big_fr_size` branch
+in this case):
+
+```dvc
+$ dvc exp branch exp-80655 big_fr_size
+Git branch 'big_fr_size' has been created from experiment 'exp-c695f'.
+To switch to the new branch run:
+
+	git checkout big_fr_size
+```
+
+Remove all the experiments that were not used:
+
+```dvc
+$ dvc exp gc --workspace --force
+```
+
+## Model checkpoints
+
+ML model checkpoints are an essential part of deep learning. ML engineers prefer
+to save the model files (or weights) at checkpoints during a training process
+and return back when metrics start diverging or learning is not fast enough.
+
+The checkpoints create a different dynamic around ML modeling process and need a
+special support from the toolset:
+
+1. Track and save model checkpoints (DVC outputs) periodically, not only the
+   final result or training epoch.
+2. Save metrics corresponding to each of the checkpoints.
+3. Reuse checkpoints - warm-start training with an existing model file,
+   corresponding code, dataset version and metrics.
+
+This new behaviour is supported in DVC 2.0. Now, DVC can version all your
+checkpoints with corresponding code and data. It brings reproducibility of DL
+processes to the next level - every checkpoint is reproducible.
+
+This is how you define checkpoints with live-metrics:
+
+```dvc
+$ dvc stage add -n train \
+        -d users.csv -d train.py \
+        -p dropout,epochs,lr,process \
+        --checkpoint model.h5 \
+        --live logs \
+    python train.py
+
+Creating 'dvc.yaml'
+Adding stage 'train' in 'dvc.yaml'
+```
+
+Note, we use `dvc stage add` command instead of `dvc run`. Starting from DVC 2.0
+we extracting all stage specific functionality under `dvc stage` unbrella.
+`dvc run` is still working but it wll be depricated in the following DVC version
+(most likely in 3.0).
+
+Start the training process and interrupt it after 5 epoches:
+
+```dvc
+$ dvc exp run
+'users.csv.dvc' didn't change, skipping
+Running stage 'train':
+> python train.py
+...
+^CTraceback (most recent call last):
+...
+KeyboardInterrupt
+```
+
+Navigate in checkpoints:
+
+```dvc
+$ dvc exp show --no-pager --no-timestamp
+┏━━━━━━━━━━━━━━━┳━━━━━━┳━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━┳━━━┳━━━━━━━━┳━━━┓
+┃ Experiment    ┃ step ┃   loss ┃ accuracy ┃ val_loss ┃ … ┃ epochs ┃ … ┃
```

Comment:
The second line looks like it has too many spaces after "Experiment" in the published version.